libyara/re_grammar.y

```diff
@@ -428,6 +428,7 @@ single
         fail_if($$ == NULL, ERROR_INSUFFICIENT_MEMORY);
 
         $$->re_class = $1;
+        $$->mask = 0xCC;
```

Comment:
I am sorry, I am updating my PR after a while, I had some deadlines, and I was also solving issues with this PR.
I changed the code based on this comment. However, I re-discovered my original motivation for this. For some reason, the type is not always correctly set up to `RE_NODE_CLASS` but `RE_NODE_LITERAL`.
This is causing missing matches on `tests/test-bitmap.c:328: rule does not match`.
The PR was, for that reason, failing tests for a few days. On other machines I have access to, the tests are OK.
I honestly have no idea why this is happening, and because I cannot recreate the failed test locally, I have trouble fixing the cause.

Reply:
Last comment regarding memory usage. I updated the `compiler.c` file with values decreasing memory usage. Hopefully, this will improve the issue you reported the last time.